diesel/src/expression/count.rs

```diff
@@ -71,3 +73,83 @@ impl<DB: Backend> QueryFragment<DB> for CountStar {
 }
 
 impl_selectable_expression!(CountStar);
+
+/// Creates a SQL `COUNT(DISTINCT ...)` expression
+///
+/// As with most bare functions, this is not exported by default. You can import
+/// it specifically as `diesel::dsl::count_distinct`, or glob import
+/// `diesel::dsl::*`
+///
+/// # Examples
+///
+/// ```rust
+/// # #[macro_use] extern crate diesel;
+/// # include!("../doctest_setup.rs");
+/// # use diesel::dsl::*;
+/// #
+/// # fn main() {
+/// #     use schema::posts::dsl::*;
+/// #     let connection = establish_connection();
+/// assert_eq!(Ok(2), posts.select(count_distinct(user_id)).first(&connection));
```

Comment:
Can we separate the assert and the query for readability?
```suggestion
/// let unique_user_count = posts.select(count_distinct(user_id)).first(&connection);
/// assert_eq!(Ok(2), unique_user_count);
```